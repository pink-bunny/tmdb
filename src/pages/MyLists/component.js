import React from 'react';
import PropTypes from 'prop-types';
import {
  Layout, Row, Col, Typography, Icon
} from 'antd';

import List from './List';

const MyLists = ({
  list,
  loading,
  error
}) => (
  <Layout>
    <Layout.Content>
      <Row>
        <Col offset={2} span={20}>
          <div className="top-margin">
            <Typography.Title>
              My Lists
              {' '}
              <Icon type="plus-circle" />
            </Typography.Title>
          </div>
        </Col>
      </Row>

      <List
        list={list}
        loading={loading}
        error={error}
      />
    </Layout.Content>
  </Layout>
);

MyLists.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired
};

export default MyLists;
