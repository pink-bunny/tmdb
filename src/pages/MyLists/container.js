import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import MyLists from './component';
import {
  fetchMyLists as fetchMyListsAction
} from '../../../state/my_lists/actions';
import {
  isMyListsLoading,
  myListsList,
  myListsListError
} from '../../../state/my_lists/selectors';

class MyListsContainer extends React.Component {
  componentDidMount() {
    const { fetchMyLists } = this.props;
    fetchMyLists();
  }

  render() {
    const {
      list,
      loading,
      error
    } = this.props;

    return (
      <MyLists
        list={list}
        loading={loading}
        error={error}
      />
    );
  }
}

MyListsContainer.propTypes = {
  fetchMyLists: PropTypes.func.isRequired,
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired
};

const mapDispatchToPtops = {
  fetchMyLists: fetchMyListsAction
};

const mapStateToProps = (state) => ({
  loading: isMyListsLoading(state),
  list: myListsList(state),
  error: myListsListError(state)
});

export default connect(mapStateToProps, mapDispatchToPtops)(MyListsContainer);
