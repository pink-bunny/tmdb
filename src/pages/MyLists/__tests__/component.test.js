import React from 'react';
import { shallow } from 'enzyme';

import MyLists from '../component';

describe('MyLists component', () => {
  const requiredProps = {
    list: [],
    loading: false,
    error: ''
  };

  it('matches snapshot', () => {
    const wrapper = shallow(<MyLists {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
