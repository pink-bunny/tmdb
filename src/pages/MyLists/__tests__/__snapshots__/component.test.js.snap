// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLists component matches snapshot 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            My Lists
             
            <WrappedComponentShell
              handleSubmitList={[MockFunction]}
              triggerComponent={[Function]}
              triggerProps={
                Object {
                  "type": "plus-circle",
                }
              }
            />
          </Title>
        </div>
      </Col>
    </Row>
    <List
      error=""
      list={null}
      loading={false}
    />
  </Adapter>
</Adapter>
`;

exports[`MyLists component received list array and matches snapshot 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            My Lists
             
            <WrappedComponentShell
              handleSubmitList={[MockFunction]}
              triggerComponent={[Function]}
              triggerProps={
                Object {
                  "type": "plus-circle",
                }
              }
            />
          </Title>
        </div>
      </Col>
    </Row>
    <List
      error=""
      list={
        Array [
          Object {
            "id": 1,
            "title": "Mock title",
          },
        ]
      }
      loading={false}
    />
  </Adapter>
</Adapter>
`;
